server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -94,7 +95,8 @@ public IndexTemplateMetaData(String name, int order, Integer version,
         this.version = version;
         this.patterns = patterns;
         this.settings = settings;
-        this.mappings = mappings;
+        this.mappings = ImmutableOpenMap.<String, CompressedXContent>builder()
+            .fPut(MapperService.SINGLE_MAPPING_NAME, mappings).build();
```

Comment:
it is not very clear, why we can't have `mappings` just as `CompressedXContent ` and have `this.mappings = mappings`?